### PR TITLE
Add fitsread and fitswrite

### PR DESCRIPTION
**Pull request by jishnub**
These convenience functions are useful while working with images. Currently I find this pattern commonly used in my code:

```julia
FITS(filename, "r") do f
    read(f[1])
end
```
this may be replaced by 
```julia
fitsread(filename)
```

and similarly for writing as well. This is easier for REPL usage too.

Not sure if these should be exported (possible conflicts in case users have defined these themselves), so currently it's up to the users to import these.

### CODE REVIEW

**Comment by mileslucas** [comment]
Thanks for this! I agree, we've been needing a quick one-liner for a while now. Some considerations:

pyfits/astropy.io.fits both have some naming conventions that we can choose to follow. For reading, they use `getdata` and for writing they use `writeto` and `append`. It would be easier for users to transition between libraries if we adopt these, although I don't find the verbage as consistent or accurate as `fitsread` and `fitswrite`. We could also consider `FITSIO.read` and `FITSIO.write`.

The other consideration is I would add an optional argument for the HDU number
```
fitsread(filename) = fitsread(filename, 1)
```
that gives a bit more flexibility.

**Review by mileslucas — APPROVED** [review]

**Comment by giordano** [comment]
Good point about having a name that this easier to guess for users coming from Astropy.

It would be good to have a docstring.  Also, export and add these functions to [`docs/src/api.md`](https://github.com/JuliaAstro/FITSIO.jl/blob/master/docs/src/api.md) if the idea is to make them "public".

**Comment by mileslucas** [comment]
For these functions, too, they should include clear documentation about the awkward dimension permutation that happens in the c/Julia interop. Someone looking for the "how do I read data" as soon as possible probably will trip over the nuance of the column-major arrays.

**Comment by jishnub** [comment]
Yes I was planning to add a docstring, didn't have time for this earlier. About the name - I chose this as [matlab](https://www.mathworks.com/help/matlab/ref/fitsread.html) uses these names, although they don't match the astropy ones. Personally I find 'getdata' to be a rather uninformative name, although I can use this if there's a consensus.

**Comment by ajwheeler** [comment]
Should there be an optional parameter to change which HDU is read for multi-extension files?  I most often need `f[2]`.

**Comment by giordano** [comment]
> Should there be an optional parameter to change which HDU is read for multi-extension files?

It's there already, `hduindex`

**Comment by ajwheeler** [comment]
🤦 , sorry.

**Review by mileslucas — APPROVED** [review]
This looks good to me, @giordano do you mind giving a look as well?

File: test/runtests.jl [hunk 1]
```diff
@@ -355,6 +355,20 @@ using Random # for `randstring`
...
+        a = ones(3,3)
+        try
+            FITS(fname, "w") do f
+                write(f, a)
+            end
+            @test FITSIO.fitsread(fname) == a
+            @test FITSIO.fitsread(fname, 1) == a
+        finally
+            rm(fname, force=true)
+        end
```
Comment (giordano) [inline test/runtests.jl:370]: If you use `mktempdir` you don't need the `try`/`catch` block:
```suggestion
        mktempdir() do dir
            fname = joinpath(dir, "file.fits")
            a = ones(3,3)
            FITS(fname, "w") do f
                write(f, a)
            end
            @test FITSIO.fitsread(fname) == a
            @test FITSIO.fitsread(fname, 1) == a
        end
```
Same thing in the test below

**Review by giordano — APPROVED** [review]
Looks good to me as well.  I only left a comment about how to simplify the tests a bit

File: test/runtests.jl (hunk 1 above)
Comment (jishnub) [reply test/runtests.jl]: Thanks for this idea! I have updated the tests to replace all the `try/catch` blocks with `mktempdir`. Other than this, file closings and cleanups are done through `do`-blocks wherever possible.

**Pull request merged**